File: product/src/main/java/com/lexuancong/product/viewmodel/product/variants/ProductVariantVm.java
package com.lexuancong.product.viewmodel.product.variants;

import com.lexuancong.product.viewmodel.image.ImageVm;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

public record ProductVariantVm(
        Long id,
        String name,
        String slug,
        String sku,
        String gtin,
        BigDecimal price,
        String avatarUrl,
        List<ImageVm>  productImages,
        Map<Long,String> optionValues

) {
}

File: frontend/models/product/variants/ProductVariantVm.ts
import {ImageVm} from "@/models/image/imageVm";

export type ProductVariantVm = {
    id : number,
    name : string,
    slug: string,
    sku: string,
    gtin: string,
    price: number,
    avatarUrl: string,
    productImages: {
        id: number,
        url: string
    }[],
    optionValues:{
        [key:number] : string
    }



}